checkpoint/side_handler.go

```diff
@@ -133,6 +133,33 @@ func PostHandleMsgCheckpoint(ctx sdk.Context, k Keeper, msg types.MsgCheckpoint,
 		return common.ErrBadBlockDetails(k.Codespace()).Result()
 	}
 
+	//
+	// Validate last checkpoint
+	//
+
+	// fetch last checkpoint from store
+	if lastCheckpoint, err := k.GetLastCheckpoint(ctx); err == nil {
```

Comment:
this is already checked in handler
why are we checking it again?

Reply:
To avoid replay since it is async. It is possible that new tx will come between handler and post handler execution.